op-chain-ops/cmd/celo-migrate/ancients.go

```diff
@@ -0,0 +1,190 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"path/filepath"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/rawdb"
+	"github.com/ethereum/go-ethereum/ethdb"
+	"github.com/ethereum/go-ethereum/log"
+	"golang.org/x/sync/errgroup"
+)
+
+// RLPBlockRange is a range of blocks in RLP format
+type RLPBlockRange struct {
+	start    uint64
+	hashes   [][]byte
+	headers  [][]byte
+	bodies   [][]byte
+	receipts [][]byte
+	tds      [][]byte
+}
+
+func migrateAncientsDb(oldDBPath, newDBPath string, batchSize uint64) (uint64, error) {
+	oldFreezer, err := rawdb.NewChainFreezer(filepath.Join(oldDBPath, "ancient"), "", false) // Can't be readonly because we need the .meta files to be created
+	if err != nil {
+		return 0, fmt.Errorf("failed to open old freezer: %w", err)
+	}
+	defer oldFreezer.Close()
+
+	newFreezer, err := rawdb.NewChainFreezer(filepath.Join(newDBPath, "ancient"), "", false)
+	if err != nil {
+		return 0, fmt.Errorf("failed to open new freezer: %w", err)
+	}
+	defer newFreezer.Close()
+
+	numAncientsOld, err := oldFreezer.Ancients()
+	if err != nil {
+		return 0, fmt.Errorf("failed to get number of ancients in old freezer: %w", err)
+	}
+
+	numAncientsNew, err := newFreezer.Ancients()
+	if err != nil {
+		return 0, fmt.Errorf("failed to get number of ancients in new freezer: %w", err)
+	}
+
+	log.Info("Ancient Block Migration Started", "process", "ancients", "startBlock", numAncientsNew, "endBlock", numAncientsOld, "count", numAncientsOld-numAncientsNew+1, "step", batchSize)
+
+	g, ctx := errgroup.WithContext(context.Background())
+	readChan := make(chan RLPBlockRange, 10)
+	transformChan := make(chan RLPBlockRange, 10)
```

Comment:
```suggestion
	readChan := make(chan RLPBlockRange)
	transformChan := make(chan RLPBlockRange)
```

I think we should make these channels unbuffered. This would reduce memory consumption and allow us to use larger batch sizes. I tested on batch sizes of 10K and 100K blocks. With the larger batch size it takes 2x less time to migrate.

Alo I ran some benchmarks and migrated 4M Alfajores blocks with the batchsize of 100K:
Buffered channels (10):
<img width="541" alt="image" src="https://github.com/celo-org/optimism/assets/20768968/669ea138-9fc8-4515-985f-9c1d37345bc8">

Unbuffered channels:
<img width="561" alt="image" src="https://github.com/celo-org/optimism/assets/20768968/10988a18-bead-4b6d-9ad8-3b7394790702">

There is no performance penalty for unbuffered channels, it even took less time for the unbuffered version (1m 5s vs 1m 20s) to migrate 4M blocks.

Reply:
Thanks for this! I've updated the script to remove the channel buffers and use a default batch size of 100k